sui-execution/latest/sui-move-natives/src/object_runtime/object_store.rs

```diff
@@ -25,11 +25,23 @@ use sui_types::{
     storage::ChildObjectResolver,
 };
 
+pub(super) struct ObjectFingerprint(Option<ObjectFingerprint_>);
+
+enum ObjectFingerprint_ {
+    Empty,
+    Original {
+        owner: ObjectID,
+        ty: MoveObjectType,
+        value: Value,
+    },
+}
```

Comment:
Dumb question: Why is this called a "fingerprint" if it stores the entire type and value of the object?

Reply:
Because the API it exposes is just "did this value change". Ideally I suppose this should really be in a separate module for privacy reasons. Let me do that. 
And blame @tzakian for the name ;)

Reply:
Yea, I think the term "fingerprint" came from the idea that you "fingerprint the value then check if the value has changed by comparing the fingerprints." The implementation here is one way of implementing fingerprints, however there are may different types "fingerprints" out there that could be used (IMO) -- all that matters is that it gives you a method to determine if the value has changed (e.g., this could ostensibly be a cryptographic hash of the value or some other more esoteric way of determining this).

All that being said though I'm not going to die on the naming hill here, so if y'all want to change it (and have a better name for it) I'm onboard with changing it (although, I really do like `fingerprint` for the name, so you would be making me sad).